src/server/pkg/deploy/assets/assets.go

```diff
@@ -59,9 +59,78 @@ type AssetOpts struct {
 	Dynamic     bool
 	EtcdNodes   int
 	EtcdVolume  string
+
 	// BlockCacheSize is the amount of memory each PachD node allocates towards
-	// its cache of PFS blocks.
+	// its cache of PFS blocks. If empty, assets.go will choose a default size.
 	BlockCacheSize string
+
+	// PachdCPUFootprint is the amount of CPU we request for each pachd node. If
+	// empty, assets.go will choose a default size.
+	PachdCPUFootprint string
+
+	// PachdNonCacheMemFootprint is the amount of memory we request for each
+	// pachd node in addition to BlockCacheSize. If empty, assets.go will choose
+	// a default size.
+	PachdNonCacheMemFootprint string
+
+	// EtcdCPUFootprint is the amount of CPU (in cores) we request for each etcd
+	// node. If empty, assets.go will choose a default size.
+	EtcdCPUFootprint string
+
+	// EtcdMemFootprint is the amount of memory we request for each etcd node. If
+	// empty, assets.go will choose a default size.
+	EtcdMemFootprint string
+}
+
+// fillDefaultResourceRequests sets any of:
+//   opts.BlockCacheSize
+//   opts.PachdNonCacheMemFootprint
+//   opts.PachdCPUFootprint
+//   opts.EtcdCPUFootprint
+//   opts.EtcdMemFootprint
+// that are unset in 'opts' to the appropriate default ('persistentDiskBackend'
+// just used to determine if this is a local deployment, and if so, make the
+// resource requests smaller)
+func fillDefaultResourceRequests(opts *AssetOpts, persistentDiskBackend backend) {
+	if persistentDiskBackend == localBackend {
+		// For local deployments, we set the resource requirements and cache sizes
+		// low so that pachyderm clusters will fit inside e.g. minikube or travis
+		if len(opts.BlockCacheSize) == 0 {
```

Comment:
The idiomatic way to check for empty string in Go is `s == ""`.  Could you do that just for the sake of consistency with the rest of the codebase?